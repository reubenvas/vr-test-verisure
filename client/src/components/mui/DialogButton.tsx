import React, { useState, useEffect } from 'react';
import Button from '@material-ui/core/Button';

type propTypes = {
    dialog: (props: { open: boolean }) => React.ReactElement;
    children: string;
};


/**
 * @param dialog is a React Dialog Element
 * @param children contains the value of the button
 * @returns a React Button which upon click will display the passed in react dialog element argument
 */
const DialogButton = ({ dialog, children }: propTypes): React.ReactElement => {
    const [dialogOpen, setDialogOpen] = useState<boolean>(false);

    const openDialog = (): void => {
        setDialogOpen(true);
    };

    useEffect(() => {
        if (dialogOpen) {
            setDialogOpen(false);
        }
    }, [dialogOpen]);

    const Dialog = dialog;

    return (
        <>
            <Button variant="contained" color="secondary" onClick={openDialog}>
                {children}
            </Button>
            <Dialog open={dialogOpen} />
        </>
    );
};

export default DialogButton;
