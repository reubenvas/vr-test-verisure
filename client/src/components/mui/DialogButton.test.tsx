import React from 'react';
import renderer from 'react-test-renderer';
import DialogButton from './DialogButton';

const MockDialog = (): React.ReactElement => (
    <div>MOCK DIALOG</div>
);

describe('DarkButton component', () => {
    test('snapshot renders', () => {
        const component = renderer.create(
            <DialogButton dialog={MockDialog}>
                Test Dialog
            </DialogButton>,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
