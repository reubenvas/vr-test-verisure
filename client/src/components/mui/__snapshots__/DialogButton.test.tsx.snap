// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkButton component snapshot renders 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Test Dialog
    </span>
  </button>,
  <div>
    MOCK DIALOG
  </div>,
]
`;
